src/bug-filing/common/markup/html-factory.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { escape } from 'lodash';
+import { MarkupFactory } from './markup-factory';
+
+const createFactory = (): MarkupFactory => {
+    const bold = (text: string): string => {
+        return `<b>${text}</b>`;
+    };
+
+    const snippet = (text: string): string => {
+        const maxSnippetLength = 256;
+        let constrainedSnippet = text;
+
+        if (text.length > maxSnippetLength) {
+            constrainedSnippet = text.substr(0, maxSnippetLength) + '...';
+        }
```

Comment:
The truncating part of this is duplicated between the two factory implementations; perhaps that belongs in a more generic string utils instead?

Reply:
Updated. Ended up using lodash truncate.